Tasks/KubernetesManifestV0/src/utils/CanaryDeploymentHelper.ts

```diff
@@ -207,17 +199,28 @@ function createCanaryObjectsArgumentString(files: string[], includeServices: boo
                 || (includeServices && helper.isServiceEntity(kind))) {
                 const canaryObjectName = getCanaryResourceName(name);
                 const baselineObjectName = getBaselineResourceName(name);
-                kindList.add(kind);
-                nameList.add(canaryObjectName);
-                nameList.add(baselineObjectName);
+                addValueToList(kindNameMap, kind, canaryObjectName);
```

Comment:
The assumption is that `canary` will always exist?

Reply:
People could accidentally use canary first too; do you want to handle that case?

Reply:
This function is called only for promote and reject. So, we can assume that canary deployment will always be present.